Compile: gcc11 emits warning message about PMIx_Get

<!--Thank you for taking the time to submit an issue!-->

## Background information

Compiling pmix 3.2.3 or github version gives compiler warning

### What version of the PMIx Reference Library are you using? <!--(e.g., v1.0, v2.1, git master @ hash, etc.)-->

3.2.3 or HEAD b1908ff8af675754f020add90bbabe4de89c3c6f

### Describe how PMIx was installed <!--(e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)-->

tarball or clone

### Please describe the system on which you are running

* Operating system/version: CentOS 8
* Computer hardware: aarch64
* Network type: IBV

-----------------------------

## Details of the problem

```.../src/osss-ucx/build/../bitbucket.org/src/shmemc/ucx/pmix_client.c:359:10: warning: ‘PMIx_Get’ reading 512 bytes from a region of size 14 [-Wstringop-overread]
  359 |     ps = PMIx_Get(&wc_pmix, PMIX_JOB_SIZE, NULL, 0, &vp);
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.../src/osss-ucx/build/../bitbucket.org/src/shmemc/ucx/pmix_client.c: In function ‘shmemc_pmi_client_init’:
.../src/osss-ucx/build/../bitbucket.org/src/shmemc/ucx/pmix_client.c:359:10: note: referencing argument 2 of type ‘const char *’
```

Looks like a type mismatch between definition of PMIx_Get in src/client and in the include file?  Or is this GCC complaining about something it shouldn't?

--- a/include/pmix.h
+++ b/include/pmix.h
@@ -201,7 +201,7 @@ PMIX_EXPORT pmix_status_t PMIx_Fence_nb(const pmix_proc_t procs[], size_t nprocs
  *     an error. The timeout parameter can help avoid "hangs" due to programming
  *     errors that prevent the target proc from ever exposing its data.
  */
-PMIX_EXPORT pmix_status_t PMIx_Get(const pmix_proc_t *proc, const char key[],
+PMIX_EXPORT pmix_status_t PMIx_Get(const pmix_proc_t *proc, const pmix_key_t key,
                                    const pmix_info_t info[], size_t ninfo,
                                    pmix_value_t **val);
 

